cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -130,13 +130,8 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 	if typedErr != nil {
 		return typedErr
 	}
-	if len(allNodes) == 0 {
-		return a.onEmptyCluster("Cluster has no nodes.", true)
-	}
-	if len(readyNodes) == 0 {
-		// Cluster Autoscaler may start running before nodes are ready.
-		// Timeout ensures no ClusterUnhealthy events are published immediately in this case.
-		return a.onEmptyCluster("Cluster has no ready nodes.", currentTime.After(a.startTime.Add(nodesNotReadyAfterStartTimeout)))
+	if a.actOnEmptyCluster(allNodes, readyNodes, currentTime) {
```

Comment:
TBH, I don't think this codepath is worth the investment. I'd rather kill it entirely (see #555). This is obviously way out of scope of this PR and I'm not asking you to fix it, just FYI

Reply:
I wasn't aware of that issue, thanks.
Since this change should not make #555 any harder, I'd suggest to keep it, the investment has already been made. If you think otherwise please let me know.

Reply:
Sure